src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,204 @@
+"""DSM external USB device data."""
+
+from __future__ import annotations
+
+from synology_dsm.api import SynoBaseApi
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoCoreExternalUSB(SynoBaseApi):
+    """Class for external USB storage devices."""
+
+    API_KEY = "SYNO.Core.ExternalDevice.Storage.USB"
+    REQUEST_DATA = {"additional": '["all"]'}
+
+    async def update(self):
+        """Updates external USB storage device data."""
+        raw_data = await self._dsm.post(self.API_KEY, "list", data=self.REQUEST_DATA)
+        if raw_data:
+            self._data = raw_data
+            if raw_data.get("data"):
+                self._data = raw_data["data"]
+
+    # Root
+    @property
+    def devices(self):
```

Comment:
```suggestion
    def get_devices(self):
```

we should always have the action in the function name (_get_xxx, set_xxx, do_xxx, ..._)
this applies for all methods

Reply:
@mib1185 
Should this apply to these as well?

In SynoCoreExternalUSBDevice:
- device_ids -> get_device_ids, 
- device_size_total -> get_device_size_total
- device_partitions -> get_device_partitions
- device_partition_ids -> get_device_partition_ids
- partitions_all_* -> get_partitions_all_*

Then in SynoUSBStoragePartition
- partition_* -> get_partition_*